### PR TITLE
ENT-9170: Added deplyment tests for built enterprise hub package

### PR DESCRIPTION
**Pull request by craigcomstock**
Can be run in docker or on a plain host (vm/metal)

Ticket: ENT-9170
Changelog: none

need changes from https://github.com/cfengine/nova/pull/2091

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thanks for submitting a pull request! Maybe @Lex-2008 can review this?

**Comment by craigcomstock** [comment]
@cf-bottom jenkins please, to check if my changes to `autogen` are "ok". :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9661/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9661/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9661/

**Comment by craigcomstock** [comment]
@cf-bottom jenkins please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9680/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9680/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9680/

File: ci/docker-build-package.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    echo "Need the ssh private key for build artifacts cache, neither env var nor mystiko was available."
+    exit 1
+  fi
+fi
+set -x # done hiding secrets
+# clean up any lingering revision file previously generated, if you are changing deps locally and iterating this is important
+[ -f "${NTECH_ROOT}/buildscripts/deps-packaging/revision" ] && rm "${NTECH_ROOT}/buildscripts/deps-packaging/revision"
+cd "${NTECH_ROOT}/buildscripts/deps-packaging"
+# see buildscripts/build-scripts/autogen for a similar workaround to ensure it stays 7 on bootstrap-oslo-dc jobs
+git config --add core.abbrev 7 # hack to match smaller commit sha on bootstrap-oslo-dc (debian-9)
```
Comment (nickanderson) [inline ci/docker-build-package.sh:40]: Could autogen write a file with what core.abbrev is set for that this could then directly use?

**Review by nickanderson — APPROVED** [review]
I don't consider my review very good, but I very much appreciate all the comments in the scripts.

File: ci/docker-build-package.sh (hunk 1 above)
Comment (craigcomstock) [reply ci/docker-build-package.sh:40]: it probably could... that would reduce the touch points :+1:

Comment (nickanderson) [reply ci/docker-build-package.sh:40]: makes sense.

File: build-scripts/autogen [hunk 2]
```diff
@@ -28,12 +28,27 @@ for p in $projects
...
 
 for i in $GITSHAOF
 do
     if [ -d $BASEDIR/$i ] && [ ! -f $BASEDIR/$i/revision ]
     then
         R=$(cd $BASEDIR/$i && git log --pretty='format:%h' -1 -- .) || false
+        (
+            cd $BASEDIR/$i
+            if ! git show $R --oneline >/dev/null; then
+                echo "abbreviated commit hash of $CORE_ABBREV is not unique. Consider increasing the value in the script $0."
```
Comment (nickanderson) [inline build-scripts/autogen:48]: Nice hint.

File: ci/docker-build-package.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+mkdir -p "${NTECH_ROOT}/cache"
+
+# pre-seed cache from sftp buildcache if possible
+# requires either environment var with private key or mystiko+pass
+eval "$(ssh-agent -s)"
+set +x # hide secrets
+if [ -n "$GH_ACTIONS_SSH_KEY_BUILD_ARTIFACTS_CACHE" ]; then
+  echo "$GH_ACTIONS_SSH_KEY_BUILD_ARTIFACTS_CACHE" | ssh-add -
+else
+  if ! pass mystiko/developers/CFEngine/jenkins/sftp-cache.sec | ssh-add -; then
```
Comment (nickanderson) [inline ci/docker-build-package.sh:30]: nifty

**Review by nickanderson — APPROVED** [review]
Can't say I fully grok it, but it looks good.

**Review by Lex-2008 — APPROVED** [review]

**Pull request merged**